client_plugin/drivers/shared/etcdops.go

```diff
@@ -72,60 +77,75 @@ type kvPair struct {
 	value string
 }
 
-// initEtcd start or join ETCD cluster depending on the role of the node
-func (d *VolumeDriver) initEtcd() error {
+type etcdKVS struct {
+	driver   *VolumeDriver
+	nodeID   string
+	nodeAddr string
+	client   *etcdClient.Client
+}
+
+// NewKvStore function: start or join ETCD cluster depending on the role of the node
+func NewKvStore(driver *VolumeDriver) *etcdKVS {
+	var e *etcdKVS
+
 	ctx := context.Background()
-	docker := d.dockerd
+	docker := driver.dockerd
 
 	// get NodeID from docker client
 	info, err := docker.Info(ctx)
 	if err != nil {
 		log.WithFields(
 			log.Fields{"error": err},
 		).Error("Failed to get Info from docker client ")
-		return err
+		return nil
 	}
 
-	// worker just returns
+	// get the swarmID and IP address of current node
 	nodeID := info.Swarm.NodeID
+	addr := info.Swarm.NodeAddr
+
+	e = &etcdKVS{
+		driver:   driver,
+		nodeID:   nodeID,
+		nodeAddr: addr,
+	}
+
+	// worker just returns
 	if info.Swarm.ControlAvailable == false {
 		log.WithFields(
 			log.Fields{"nodeID": nodeID},
 		).Info("Swarm node role: worker. Action: return from InitEtcd ")
```

Comment:
Thanks for catching the function name. This one in fact is not a error message but a log info for debugging use.